electrum/lnpeer.py

```diff
@@ -955,6 +1053,13 @@ async def trigger_force_close(self, channel_id: bytes):
             your_last_per_commitment_secret=0,
             my_current_per_commitment_point=latest_point)
 
+    async def try_force_closing(self, channel_id: bytes):
+        channels_with_peer = list(self.channels.keys())
+        channels_with_peer.extend(self.temp_id_to_id.values())
+        if channel_id not in channels_with_peer:
+            raise ValueError(f"channel {channel_id.hex()} does not belong to this peer")
+        await self.lnworker.try_force_closing(channel_id)
```

Comment:
Note that I've just changed the `lnworker.try_force_closing` API on master in https://github.com/spesmilo/electrum/commit/556b98736e78eb04bc68fd44e701e3fa174023a9.
It is no longer async, which means this method does not need to be async either, and they can be called easily from e.g. `on_error` without that needing to be async.
The issue with `try_force_closing` being async and awaiting it (or even if you just call `asyncio.ensure_future(try_force_closing())`) is that it takes a few event loop iterations for it to actually execute and in particular to set the channel state. In case of calling `Peer.try_force_closing()` as-is in the PR currently, it takes at least two event loop iterations.
During that time, the `_message_loop` might process further messages potentially affecting the channel. It seems difficult to reason about this, so it is best to just avoid the situation in the first place by setting the channel state immediately. `lnworker.schedule_force_closing` does this.

Reply:
Yes, that's safer